internal/services/controller/controller.go

```diff
@@ -103,6 +104,11 @@ func New(
 			f.Autoscaling().V1().HorizontalPodAutoscalers().Informer()
 	}
 
+	if v.MinorInt() >= 21 {
```

Comment:
Imho this is a bad pattern we started using some time ago, instead we can just use the discovery API and check if the objects are registered.

Reply:
(See pod metrics)

Reply:
Not sure if it's worth tackling right now, as the version is well known 🤔
Also, we actually are not checking for a major version, so our agent oob does not support k8s 2.0.0 😿